zypp/HistoryLog.cc

```diff
@@ -334,6 +334,27 @@ namespace zypp
     }
   }
 
+  void HistoryLog::patchStateChange( const PoolItem & pi, const std::string &oldstate )
+  {
+    const Patch::constPtr p = asKind<Patch>(pi.resolvable());
+    if (!p)
+      return;
```

Comment:
This should work: `if ( ! pi.isKind<Patch>() ) return;`
The _'conversion to pointer and cast'_ style test is how we did it ~10 years ago :)

If there are more ancient test styles in the file, yoi can fix them on the fy too.

Reply:
Ok, but it won't make any difference in lines though because i need to cast to Patch anyway ;)